From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: clavin <clavin@electronjs.org>
Date: Sat, 13 Dec 2025 15:30:46 -0800
Subject: expose GTK UI platform field

Chromium used to expose this as a public static method, backed by global storage. In 7237910, these conveniences were removed.

This patch should be proposed upstream.

Ref: https://chromium-review.googlesource.com/c/chromium/src/+/7237910 "7237910: Remove g_gtk_ui global"

diff --git a/ui/gtk/gtk_ui.h b/ui/gtk/gtk_ui.h
index f1110e770b5047277a62389f297b8ba54f2f6cbc..8234ed298f05dafb5e3de734502525dafa7241d3 100644
--- a/ui/gtk/gtk_ui.h
+++ b/ui/gtk/gtk_ui.h
@@ -53,6 +53,8 @@ class GtkUi : public ui::LinuxUiAndTheme {
 
   ~GtkUi() override;
 
+  GtkUiPlatform* GetPlatform() const { return platform_.get(); }
+
   // Setters used by SettingsProvider:
   void SetWindowButtonOrdering(
       const std::vector<views::FrameButton>& leading_buttons,
